### DISCUSSION TITLE
Visualizing the github object

### DISCUSSION THREAD

**Discussion by aleks-ivanov** [community/community#27206] [Actions]
<p>I come from JS development and I have a habit of digging, inside the dev tools, into objects to see their props and methods, so that I could use them for whatever functionality I wanted to create.</p>
<p>In GHActions, I understood that there is this workflow context object github, which encompasses everything that happens like events, repo data etc.</p>
<p>Currently, I am trying to create a logic for some of the steps inside a workflow and I would like to if I can base it on information the github object can provide.</p>
<p>Is there any, hopefully as straightforward as objects in dev tools, way to visualize and rummage through the whole github object, optimally in real time ?</p>

**Comment by dentuzhik** [comment]
<p>I know that this likely doesn’t answer your question, but I am not aware of such tool existing, but you can always write a small node script which logs it out. Probably you could also utilise Github’s toolkit for that: <a href="https://github.com/actions/toolkit/blob/8d11ee5a8c46598f9c08f0905995c84f3352cff0/packages/github/src/context.ts#L6" rel="noopener nofollow ugc">https://github.com/actions/toolkit/blob/8d11ee5a8c46598f9c08f0905995c84f3352cff0/packages/github/src/context.ts#L6</a></p>
<p>On the other hand documentation on the context is rather solid, and we rarely had to dig into internals to find out what it’s composed from:<br>
</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#contexts" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#contexts" target="_blank" rel="noopener nofollow ugc">Context and expression syntax for GitHub Actions - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#contexts" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Maybe you could describe your usecase a bit better?</p>

**Comment by aleks-ivanov** [comment]
<p>How would you go about doing this ? - “small node script which logs it out”</p>
<p>So far, sounds like this would solve my conundrum, the best.</p>
<p>As far as particular usecase, I was thinking more in general, but currently I am trying to solve this and couldn’t figure it out using the docs myself:</p>
<aside class="quote" data-post="1" data-topic="148762">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/distinguishing-between-prs/148762">Distinguishing between PRs</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    I would like a certain step in my workflow to execute only on PRs from currentRepo/feature/branch to currentRepo/master (currentRepo is a fork!) and NOT EXECUTE on PRs from currentRepo/feature/branch to upstream/master. 
Is there a way to distinguish between the two PRs ?
  </blockquote>
</aside>

**Comment by dentuzhik** [comment, marked as answer]
<p>So as I mentioned above you can use <code>actions/toolkit</code> to access github context in node, that’s one way:<br>
</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/actions/toolkit#typescript-action-walkthrough" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:400/400;"><img src="https://avatars2.githubusercontent.com/u/44036562?s=400&amp;amp;v=4" class="thumbnail" width="400" height="400"></div>

<h3><a href="https://github.com/actions/toolkit#typescript-action-walkthrough" target="_blank" rel="noopener nofollow ugc">actions/toolkit - TypeScript Action Walkthrough</a></h3>


  <p><span class="label1">The GitHub ToolKit for developing GitHub Actions. Contribute to actions/toolkit development by creating an account on GitHub.</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Then to make this even more lo-key investment and don’t do bundling you could use <code>acitons/github-script</code> to access toolkit inline inside yaml, as you can see both <code>github</code> and <code>context</code> are parts of the arguments:<br>
</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/actions/github-script" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:400/400;"><img src="https://avatars2.githubusercontent.com/u/44036562?s=400&amp;amp;v=4" class="thumbnail" width="400" height="400"></div>

<h3><a href="https://github.com/actions/github-script" target="_blank" rel="noopener nofollow ugc">actions/github-script</a></h3>


  <p><span class="label1">Write workflows scripting the GitHub API in JavaScript</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>When it comes to your specific usecase, I believe you need an “if” with particular assertions agaist<br>
<code>github.head_ref</code> which represents the “source” of PR and <code>github.base_ref</code> which represents the target.</p>
<p>Docs on “if”:<br>
</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif" target="_blank" rel="noopener nofollow ugc">Workflow syntax for GitHub Actions - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>You might also want to have a look at available functions for expressions inside “if” statements, in this case <code>contains</code> seems to be appropriate:<br>
</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#functions" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#functions" target="_blank" rel="noopener nofollow ugc">Context and expression syntax for GitHub Actions - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#functions" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
